### ISSUE TITLE
[Request] log filament usage of every user

### ISSUE THREAD

**Issue by n-dim** [foosel/OctoPrint#652]
Having the user management in OcotoPrint it would be great to track the filament usage on a per user base. This way in our fablab we could account the filament usage of different people very easily. The problem is often, if you stop a print midway, you don't know how much filament was used so far. But OctoPrint could track this by constantly logging the executed extrusion length, right?

**Comment by stv0g** [comment]
Just voting for this feature :+1: 

And maybe we could a cost estimation for filament too? Its just a rough estimation. But taking the filament length and width we could calculate the volume/weight.

**Comment by JoseAlcerreca** [comment]
+1, but no need to estimate cost in my opinion.

**Comment by coniferous1** [comment]
+1 Cost would eventually be nice, but I think just keeping track of what every user is using is primary. Cost could come later.

**Comment by n-dim** [comment]
I think once the usage is tracked a cost calculation is just a single input field in the settings and a simple multiplication with the usage. I think the usage calculation is what is complicated about this feature. Once this is done the cost calculation would not be hard to implement. But overall I'm just guessing.

**Comment by miton18** [comment]
+1 very interesting feature, just lenght filement per user

**Comment by thirtythreeforty** [comment]
I would like this feature as well, mainly just the per-user logging.  In addition, I think it would be nice to have an option to store the cost of filament (like @n-dim said -- it should not be hard), so that the filament estimate metric could include the cost of the print, like this:

![index4](https://cloud.githubusercontent.com/assets/3952613/7507587/84635a92-f437-11e4-8bbf-288c3b957e6f.png)

I know that my users don't care about how much filament a print will take, but they might care about how much it costs, especially if their approach is "measure once, print twice."

**Comment by thirtythreeforty** [comment]
@foosel I'm interested in writing a logging system like this one.  Do you have an opinion on how it should store logged data on the backend?  It would be nice to back everything with a database so we could run queries such as "who printed things on 9 May 2015?" or "how much filament has user *X* used?".

**Comment by arekm** [comment]
Looks like this requests improved version of https://github.com/imrahil/OctoPrint-PrintHistory (missing per user info and actual filament usage)

**Comment by foosel** [comment]
User who started a print will be available starting with 1.3.9. Everything else is will be up to plugin authors 
 utilizing that information.

**Comment by foosel** [comment]
Forgot to close this on release of 1.3.9 which added the data needed for plugins to implement this.
